examples/06-multiple-connections/client.c

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright 2020, Intel Corporation */
+
+/*
+ * client.c -- a client of the multiple-connections example
+ *
+ * Please see README.md for a detailed description of this example.
+ */
+
+#include <librpma.h>
+#include <inttypes.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include "common.h"
+
+static char *Names[] = {
+	"Oksana",
+	"Lukasz",
+	"Tomasz",
+	"Janek",
+	"Chet"
+};
+
+#define NAMES_NUM (sizeof(Names) / sizeof(Names[0]))
+
+#define USAGE_STR "usage: %s <server_address> <service>\n"
+
+int
+main(int argc, char *argv[])
+{
+	/* validate parameters */
+	if (argc < 3) {
+		fprintf(stderr, USAGE_STR, argv[0]);
+		exit(-1);
+	}
+
+	/* read common parameters */
+	char *addr = argv[1];
+	char *service = argv[2];
+	int ret, ret2;
+
+	/* RPMA resources - general */
+	struct rpma_peer *peer = NULL;
+	struct rpma_conn *conn = NULL;
+
+	/* resources - memory region */
+	void *mr_ptr = NULL;
+	struct rpma_mr_local *mr = NULL;
+
+	/*
+	 * lookup an ibv_context via the address and create a new peer using it
+	 */
+	ret = client_peer_via_address(addr, &peer);
+	if (ret)
+		return ret;
+
+	/* allocate a memory */
+	mr_ptr = malloc_aligned(DWORD);
+	if (mr_ptr == NULL) {
+		ret = -1;
+		goto err_peer_delete;
+	}
+
+	/* pick a name */
+	srand(time(0) + getpid());
+	const char *name = Names[(long unsigned int)rand() % NAMES_NUM];
+	(void) strncpy((char *)mr_ptr, name, DWORD);
+	printf("My names is: %s\n", (char *)mr_ptr);
+
+	/* register the memory */
+	ret = rpma_mr_reg(peer, mr_ptr, DWORD, RPMA_MR_USAGE_READ_SRC,
+			RPMA_MR_PLT_VOLATILE, &mr);
+	if (ret) {
+		print_error_ex("rpma_mr_reg", ret);
+		goto err_mr_free;
+	}
+
+	struct rpma_conn_private_data pdata;
+	rpma_mr_descriptor desc;
+	pdata.ptr = &desc;
+	pdata.len = sizeof(rpma_mr_descriptor);
+
+	/* receive the memory region's descriptor */
+	ret = rpma_mr_get_descriptor(mr, &desc);
+	if (ret) {
+		print_error_ex("rpma_mr_get_descriptor", ret);
+		goto err_mr_dereg;
+	}
+
+	/* establish a new connection to a server listening at addr:service */
+	ret = client_connect(peer, addr, service, &pdata, &conn);
+	if (ret)
+		goto err_mr_dereg;
+
+	ret = common_wait_for_conn_close_and_disconnect(&conn);
```

Comment:
I think examples should be self-contained. Any 'utils' or 'common' things, if neccessery, should only be for non-core things.